packages/@ourworldindata/grapher/src/entitySelector/EntitySelector.tsx

```diff
@@ -216,20 +235,121 @@ export class EntitySelector extends React.Component<{
         return this.manager.entitySelectorState.localEntityNames
     }
 
+    @computed private get populationColumn(): CoreColumn | undefined {
+        return this.manager.entitySelectorState.populationColumn
+    }
+
+    @computed private get gdpPerCapitaColumn(): CoreColumn | undefined {
+        return this.manager.entitySelectorState.gdpPerCapitaColumn
+    }
+
+    @computed private get isLoadingExternalSortColumn(): boolean {
+        return (
+            this.manager.entitySelectorState.isLoadingExternalSortColumn ??
+            false
+        )
+    }
+
     @computed private get table(): OwidTable {
         return this.manager.tableForSelection
     }
 
-    @computed private get sortColumns(): CoreColumn[] {
-        const activeSlugs = this.manager.activeColumnSlugs ?? []
+    @computed private get hasCountryOrRegionEntities(): boolean {
+        return (
+            intersection(
+                this.availableEntityNames,
+                regions.map((region) => region.name)
+            ).length > 0
+        )
+    }
 
-        const sortColumns = activeSlugs
+    @computed private get numericalChartColumns(): CoreColumn[] {
+        const activeSlugs = this.manager.activeColumnSlugs ?? []
+        return activeSlugs
             .map((slug) => this.table.get(slug))
             .filter(
                 (column) => column.hasNumberFormatting && !column.isProjection
             )
+    }
+
+    @computed
+    private get numericalChartColumnsWithoutPopulationAndGdpPerCapita(): CoreColumn[] {
+        return this.numericalChartColumns.filter(
+            (column) =>
+                column.slug !== this.populationSlug &&
+                column.slug !== this.gdpPerCapitaSlug
+        )
+    }
+
+    @computed private get sortColumns(): CoreColumn[] {
+        return excludeUndefined([
+            this.table.entityNameColumn,
+            this.populationColumn,
+            this.gdpPerCapitaColumn,
+            ...this.numericalChartColumnsWithoutPopulationAndGdpPerCapita,
+        ])
+    }
+
+    @computed private get populationColumnUsedInChart():
+        | CoreColumn
+        | undefined {
+        const activePopulationColumn = this.numericalChartColumns.find(
+            (column) =>
+                isPopulationVariableId(column.slug) ||
+                !!makeColumnLabel(column).match(/\bpopulation\b/i)
```

Comment:
Hm, this regex will also match "Population density", "Female population", "Share of population in poverty", etc.

Reply:
For population, I think it is best to rely on `isPopulationVariableId` only and not use a regex since we'd get many spurious matches ([Datasette](https://datasette-public.owid.io/owid?sql=select%0D%0A++distinct+v.name,%0D%0A++v.id%0D%0Afrom%0D%0A++chart_dimensions+cd%0D%0A++join+variables+v+on+cd.variableId+%3D+v.id%0D%0Awhere%0D%0A++v.name+like+'%25population%25'))

Reply:
Yes, agreed.
Or otherwise, match only `/Population( \(historical estimates\))?/`, which currently finds exactly these four.